File: leetcode/349. Intersection of Two Arrays/main.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> s1 = new HashSet<>();
        Set<Integer> s2 = new HashSet<>();

        for (int val : nums1) {
            s1.add(val);
        }
        for (int val : nums2) {
            s2.add(val);
        }


        s1.retainAll(s2);
        int[] result = new int[s1.size()];

        int i = 0;
        for (int val : s1) {
            result[i++] = val;
        }

        return result;
    }
}

File: leetcode/349. Intersection of Two Arrays/main.ts
const intersection = (nums1: number[], nums2: number[]): number[] => {
    const [s1, s2] = [new Set(nums1), new Set(nums2)];

    let result = [] as number[];
    s1.forEach((val) => {
        if (s2.has(val)) {
            result.push(val);
        }
    })

    return result;
};
